core/src/main/java/tech/pegasys/web3signer/core/service/JacksonSerializers.java

```diff
@@ -0,0 +1,133 @@
+/*
+ * Copyright 2020 ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+package tech.pegasys.web3signer.core.service;
+
+import tech.pegasys.web3signer.core.multikey.metadata.SigningMetadataException;
+
+import java.io.IOException;
+import java.math.BigInteger;
+
+import com.fasterxml.jackson.core.JsonGenerator;
+import com.fasterxml.jackson.core.JsonParser;
+import com.fasterxml.jackson.databind.DeserializationContext;
+import com.fasterxml.jackson.databind.JsonDeserializer;
+import com.fasterxml.jackson.databind.JsonSerializer;
+import com.fasterxml.jackson.databind.SerializerProvider;
+import org.apache.tuweni.bytes.Bytes;
+import org.apache.tuweni.units.bigints.UInt64;
+
+public class JacksonSerializers {
+
+  public static class Base64Deserialiser extends JsonDeserializer<Bytes> {
+
+    @Override
+    public Bytes deserialize(final JsonParser p, final DeserializationContext ctxt) {
+      try {
+        return Bytes.fromBase64String(p.getValueAsString());
+      } catch (final Exception e) {
+        throw new SigningMetadataException("Invalid hex value for private key", e);
+      }
+    }
+  }
+
+  public static class Base64Serialiser extends JsonSerializer<Bytes> {
+    @Override
+    public void serialize(
+        final Bytes value, final JsonGenerator gen, final SerializerProvider serializers)
+        throws IOException {
+      gen.writeString(value.toBase64String());
+    }
+  }
+
+  public static class HexDeserialiser extends JsonDeserializer<Bytes> {
```

Comment:
won't it conflict with Base64Deserialiser? both of them are converting Bytes?

Reply:
Only if you use both at the same time. This class doesn't register any of the deserializers it's merely a factory, that is up to the Jackson module.